Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1497,7 +1488,368 @@ The hooked function outputted the decrypted string. You extracted the secret str
+
+###### Memory Dump
+
+Wether you are using a rooted phone with frida-server installed or a non-rooted device, you can dump the app's process memory with [objection](https://github.com/sensepost/objection "Objection") and [Fridump](https://github.com/Nightbringer21/fridump "Fridump"). To take advantage of these tools on a non-rooted device, the Android app must be repackaged with `frida-gadget.so` and re-signed. A detailed explanation of this process is in the section "[Dynamic Analysis on Non-Rooted Devices](#dynamic-analysis-on-non-rooted-devices "Dynamic Analysis on Non-Rooted Devices").
+
+With objection it is possible to dump all memory of the running process on the device by using the command `memory dump all`.
+
+```shell
+$ objection --gadget sg.vantagepoint.helloworldjni explore
+
+sg.vantagepoint.helloworldjni on (google: 8.1.0) [usb] # memory dump all /Users/foo/memory_Android/memory
+
+Will dump 719 rw- images, totalling 1.6 GiB
+Dumping 1002.8 MiB from base: 0x14140000  [------------------------------------]    0%  00:11:03(session detach message) process-terminated
+Dumping 8.0 MiB from base: 0x7fc753e000  [####################################]  100%
+Memory dumped to file: /Users/foo/memory_Android/memory
+```
+
+In this case there was an error but it could obtain part of the binary. We are able to find the "Hello from C++" strings with radare2:
+
+```bash
+$ r2 /Users/foo/memory_Android/memory
+[0x00000000]> izz~Hello from
+1136 0x00065270 0x00065270  14  15 () ascii Hello from C++
+```
+
+Alternatively you can use Fridump. This time, we will input a string and see if we can find it in the memory dump. For this, open the [MSTG Hacking Playground](https://github.com/OWASP/MSTG-Hacking-Playground/tree/master/Android "MSTG Hacking Playground") app, navigate to "OMTG_DATAST_002_LOGGING" and enter "owasp-mstg" to the password field. Next, run Fridump:
+
+```bash
+python3 fridump.py -U sg.vp.owasp_mobile.omtg_android -s
+
+Current Directory: /Users/foo/git/fridump
+Output directory is set to: /Users/foo/git/fridump/dump
+Starting Memory dump...
+Oops, memory access violation!-------------------------------] 0.28% Complete
+Progress: [##################################################] 99.58% Complete
+Running strings on all files:
+Progress: [##################################################] 100.0% Complete
+
+Finished!
+```
+
+It will take a while until it's completed (you might get several memory access violation errors) and you'll get a collection of *.data files inside the dump folder. When you add the `-s` flag, all strings are extracted from the dumped raw memory files and added to the file `strings.txt`, which is also stored in the dump directory.
```

Comment:
what do these violation errors mean?

Reply:
both this and the previous are related. I've added a comment until I have more concrete details or explanation. For now it is safe to ignore and the user doesn't have to debug it or fix it. It's an issue in the Frida agent trying to read where it's not allowed to.